### PR TITLE
feat: Update pipeline templates and bumb haystack version to 1.12.2

### PR DESCRIPTION
**Pull request by erendabanlioglu**

### CODE REVIEW

File: pipelines/SparseDocSearch_BM25.yaml [hunk 1]
```diff
@@ -15,7 +14,7 @@ components:
   - name: DocumentStore
     type: DeepsetCloudDocumentStore # This is the only supported document store in deepset Cloud
   - name: Retriever # Selects the most relevant documents from the document store
-    type: BM25Retriever # Sparse retriever
+    type: ElasticsearchRetriever # Sparse retriever
```
Comment (ju-gu) [inline pipelines/SparseDocSearch_BM25.yaml:17]: Type here should be "BM25Retriever" as for the others

**Review by ju-gu — APPROVED** [review]
Just some little changes, otherwise lgtm

File: pipelines/SparseDocSearch_BM25.yaml (hunk 1 above)
Comment (erendabanlioglu) [reply pipelines/SparseDocSearch_BM25.yaml]: updated, can you verify?

**Comment by ju-gu** [comment]
ok, can be merged I would say

**Comment by agnieszka-m** [comment]
@mathislucka regarding your comment about:

> # With a dense retriever, it's good to split your files into smaller documents

So actually just realized that's not true, I think. Preprocessor takes in documents and outputs documents as well. There's text converter and pdf converter before it that change files to documents, then preprocessor takes in these documents and splits them into smaller documents. 
Can you confirm?

**Review by agnieszka-m — APPROVED** [review]

**Comment by mathislucka** [comment]
> @mathislucka regarding your comment about:
> 
> > With a dense retriever, it's good to split your files into smaller documents
> 
> So actually just realized that's not true, I think. Preprocessor takes in documents and outputs documents as well. There's text converter and pdf converter before it that change files to documents, then preprocessor takes in these documents and splits them into smaller documents. Can you confirm?

Yes, technically true, but the user doesn't really know what the file converter does and what goes into the pipeline is files. The documents created from the converter are also just the full files.

**Comment by erendabanlioglu** [comment]
Tested all 5 templates, they are indexed and search query works ✅ 

Screenshot 2023-01-18 at 12 53 45

**Comment by erendabanlioglu** [comment]
> Yes, technically true, but the user doesn't really know what the file converter does and what goes into the pipeline is files. The documents created from the converter are also just the full files.

It says it above in text and pdf converter that their files converted to documents `name: TextConverter # Converts TXT files into documents`. Either we keep as it or change converters description.

**Comment by agnieszka-m** [comment]
I'd keep it as is. In documentation, we say that preprocessor takes in documents and spits out documents. I think it would be confusing to suddenly talk about files here

**Comment by erendabanlioglu** [comment]
I'm merging it as it is, please open other pr if necessary

**Pull request merged**